libbeat/outputs/output_reg.go

```diff
@@ -47,14 +47,37 @@ type IndexSelector interface {
 	Select(event *beat.Event) (string, error)
 }
 
+type IGroup interface {
```

Comment:
This naming is just for the sake of the POC. Happy to change it if things seem to be heading in the right direction in this PR.